design/one-pager-package-image-config.md

```diff
@@ -0,0 +1,377 @@
+
+### API
+
+```yaml
+apiVersion: pkg.crossplane.io/v1beta1
+kind: ImageConfig
+metadata:
+  name: acme-packages
+spec:
+  matchImages:
+    - prefix: registry1.com/acme-co/configuration-foo
+    - prefix: registry1.com/acme-co/configuration-bar
+    - prefix: registry1.com/acme-co/function-baz
+  registry:
+    authentication:
+      pullSecretRef:
+        name: acme-registry-credentials
+    tls:
+      enabled: true
+      insecure: false
+      caBundleConfigMapRef:
+        name: registry1-ca
+        key: ca.crt
+  verification:
+    provider: cosign
+    cosign:
+      authorities:
+        - name: verify acme builds
+          keyless:
+            url: https://fulcio.sigstore.dev
+            identities:
+              - issuer: https://token.actions.githubusercontent.com
+                subjectRegExp: https://github.com/acme-co/crossplane-packages/*
+          attestations:
+            - name: verify attestations
+              predicateType: spdxjson
+```
+
+The `ImageConfig` API `spec` has the following fields:
+
+- `matchImages`: A list of image prefixes to match. The package manager will
+  apply the settings defined in the `ImageConfig` object to the images that
+  match the prefixes. The longest prefix match will be selected in case of
+  multiple matches.
+- `registry`: Configuration for interacting with the registry hosting the images.
+  - `authentication`: Authentication settings for the registry.
+    - `pullSecretRef`: Reference to the Kubernetes Secret containing the
+      credentials to pull images from the registry. This secret must be of type
+      [`kubernetes.io/dockerconfigjson`].
+  - `tls`: TLS settings for the registry.
+    - `enabled`: Whether to enable TLS for the registry. Default is `true`.
+    - `insecure`: Whether to skip verifying the server's certificate. Default is
+      `false`.
+    - `caBundleConfigMapRef`: Reference to the ConfigMap containing the CA
+      certificate bundle to use for verifying the server's certificate.
+- `verification`: Configuration for verifying image signatures.
+  - `provider`: The provider to use for verifying image signatures. In the
+    beginning, only `cosign` will be supported.
+  - `cosign`: Configuration for verifying images using cosign.
+    - `authorities`: List of authorities to use for verifying images.
+      - `name`: The name of the authority.
+      - `keyless`: Keyless verification settings.
+        - `url`: The URL of the keyless authority.
+        - `identities`: List of identities to use for verification.
+          - `issuer`: The issuer of the identity.
+          - `subjectRegExp`: The regular expression to match the subject of the
+            identity.
+      - `attestations`: List of attestations to use for verification.
+        - `name`: The name of the attestation.
+        - `predicateType`: The type of the predicate to use for verification.
+
+The `ImageConfig` object will be a global object in the cluster, configuring
```

Comment:
Nit: Maybe "apply globally"? This confused me for a second - I thought it might have meant it was cluster scoped, or a singleton.

It did make more sense to me once I read the rest of the paragraph though so feel free to ignore.